core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala

```diff
@@ -176,7 +176,13 @@ private[ui] class MasterPage(parent: MasterWebUI) extends WebUIPage("") {
   private def workerRow(worker: WorkerInfo): Seq[Node] = {
     <tr>
       <td>
-        <a href={worker.webUiAddress}>{worker.id}</a>
+        {
+          if (parent.master.reverseProxy) {
```

Comment:
This pattern is being used in a lot of places and it's kinda becoming noisy... perhaps a better approach would be to have a method somewhere that takes both the proxied and non-proxied versions, and returns which one to use. e.g.

    def href(normal: String, proxied: String): String = if (reverseProxy) s"/proxy/$proxied" else normal